### PR TITLE
Feature/decrypt data

### PR DESCRIPTION
**Pull request by mlobac**
## What does this PR do?

Decrypt data from backend

- [+] Checked on iOS
- [+] Checked on Android

### CODE REVIEW

File: class/abstract-hd-segwit-p2sh-wallet.js [hunk 1]
```diff
@@ -180,6 +180,36 @@ export class AbstractHDSegwitP2SHWallet extends AbstractHDWallet {
...
 
+  async getXpriv() {
+    if (this._xpriv) {
+      return this._xpriv; // cache hit
+    }
+    // first, getting xpub
+    this.seed = await this.getSeed();
+    const root = HDNode.fromSeed(this.seed, config.network);
+    const path = this._getPath();
+    const child = root.derivePath(path);
```
Comment (aleparty) [inline class/abstract-hd-segwit-p2sh-wallet.js]: since we are reusing the same for getKeyPair maybe we could add and import it from helpers?

**Comment by aleparty** [comment]
maybe we could add some unit test to that as well?

File: class/abstract-hd-segwit-p2sh-wallet.js (hunk 1 above)
Comment (mlobac) [reply class/abstract-hd-segwit-p2sh-wallet.js]: moved reusable code to separate method

**Comment by mlobac** [comment]
> maybe we could add some unit test to that as well?

added unit test to check if encryption/decryption work with current wallet key pair

**Review by aleparty — APPROVED** [review]
LGTM

**Pull request merged**